### ISSUE TITLE
Untested Theory: FormField::getAttributesHTML() needs casting to HTMLText

### ISSUE THREAD

**Issue by silbinarywolf** [silverstripe/silverstripe-framework#8036]
## Affected Version

- SilverStripe 4.1 and lower.

## Description

I'm pretty sure if you try passing `getAttributesHTML` from one template to another, you'll end up with quote issues (due to double escaping). I need to test this out, but because this directly affects my Components module, I'll most likely put together a PR if this is the case.

**Comment by tractorcow** [comment]
It's already cast to HTML fragment.

```php
private static $casting = array(
        // ...
        'AttributesHTML' => 'HTMLFragment', // property $AttributesHTML version
        'getAttributesHTML' => 'HTMLFragment', // method $getAttributesHTML($arg) version
        // ...
);
```

Can you demonstrate how this can be double escaped?

**Comment by silbinarywolf** [comment]
I'll hopefully be investigating later today, so I'll close the issue or post back with more useful information :)

**Comment by silbinarywolf** [comment]
Will investigate today.

**Comment by silbinarywolf** [comment]
@tractorcow 

Nope, its just my problem!
My component module output concats unnecessarily, which makes it lose escaped values and become cast to a string.

ie. https://github.com/silbinarywolf/silverstripe-components/blob/master/docs/en/advanced-usage.md#example-template-cache-output

**Comment by silbinarywolf** [comment]
Problem came with coding this. Due to the casting changes made, I wasn't aware that `AttributesHTML` will not cast properly, so only the former of the two work properly as expected in templates.
```
<% include ButtonInclude testName="func", attributesHTML=$getAttributesHTML("class","type") %>
<% include ButtonInclude testName="non-func", attributesHTML=$attributesHTML %>
```

**Comment by silbinarywolf** [comment]
Fixed stuff on my side now.
Basically the problem is my module techically allows passing a variable + adding additional string data, so I had to get creative with the parser.

My solution to my problem is here. Might help you guys if you ever decide to allow syntax like this:
```
<% include ButtonInclude testName="non-func", attributesHTML="<% if $attributesHTML %>$attributesHTML<% end_if %> more-string" %>
```

Solution:
https://github.com/silbinarywolf/silverstripe-components/issues/11#issuecomment-385231954
